.github/workflows/main.yml

```diff
@@ -39,3 +39,20 @@ jobs:
       - uses: hadolint/hadolint-action@v1.5.0
         with:
           failure-threshold: error
+
+  docker-publish:
+    if: github.ref == 'refs/heads/main'
+    runs-on: ubuntu-latest
+    needs: [test, lint-dockerfile]
+    env:
+      image: ghcr.io/opensafely-core/cohortextractor-v2:latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Pull image
+        run: docker pull ${{ env.image }}
+      - name: Build image
+        run: docker build . --file Dockerfile --tag ${{ env.image }}
+      - name: Log into GitHub Container Registry
+        run: docker login https://ghcr.io -u ${{ github.actor }} --password ${{ secrets.DOCKER_RW_TOKEN }}
```

Comment:
I'll need to enable this secret for this repo, will do so now

Reply:
Oh, thanks.

Reply:
Done